packages/next/lib/eslint/runLintCheck.ts

```diff
@@ -40,109 +80,139 @@ async function lint(
       eventInfo: EventLintCheckCompleted
     }
 > {
-  // Load ESLint after we're sure it exists:
-  const mod = await import(deps.resolved.get('eslint')!)
+  try {
+    // Load ESLint after we're sure it exists:
+    const deps = await hasNecessaryDependencies(baseDir, requiredPackages)
 
-  const { ESLint } = mod
-  let eslintVersion = ESLint?.version
+    if (deps.missing.some((dep) => dep.pkg === 'eslint')) {
+      Log.error(
+        `ESLint must be installed${
+          lintDuringBuild ? ' in order to run during builds:' : ':'
+        } ${chalk.bold.cyan(
+          isYarn(baseDir)
+            ? 'yarn add --dev eslint'
+            : 'npm install --save-dev eslint'
+        )}`
+      )
+      return null
+    }
+
+    const mod = await import(deps.resolved.get('eslint')!)
 
-  if (!ESLint) {
-    eslintVersion = mod?.CLIEngine?.version
+    const { ESLint } = mod
+    let eslintVersion = ESLint?.version
+
+    if (!ESLint) {
+      eslintVersion = mod?.CLIEngine?.version
+
+      if (!eslintVersion || semver.lt(eslintVersion, '7.0.0')) {
```

Comment:
Should this version check happen on line 118 instead? Because line 103 assigns the version but doesn't check semver if it existed there.

Reply:
Ah yeah just cleaned it up a bit and removed the initial check for `ESLint` and the unnecessary 2X assignment of `eslintVersion`